### ISSUE TITLE
Error in dependences 

### ISSUE THREAD

**Issue by alexmeseguer** [flutter/udacity-course#137]
I have an error when I run an Android Devices, but it's working correctly with iOS Devices. 

I just to start my app without modifies.

This error:

Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...
Resolving dependencies...
* Error running Gradle:
Exit code 1 from: /Users/alejandromeseguersilvestre/Desktop/Alumbrado/alumbrado_flutter/alumbrado_flutter_app/android/gradlew app:properties:
Project evaluation failed including an error in afterEvaluate {}. Run with --stacktrace for details of the afterEvaluate {} error.

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/alejandromeseguersilvestre/Desktop/Alumbrado/alumbrado_flutter/alumbrado_flutter_app/android/app/build.gradle' line: 25

* What went wrong:
A problem occurred evaluating project ':app'.
> Could not resolve all files for configuration 'classpath'.
   > Could not find lint-gradle-api.jar (com.android.tools.lint:lint-gradle-api:26.1.2).
     Searched in the following locations:
         https://jcenter.bintray.com/com/android/tools/lint/lint-gradle-api/26.1.2/lint-gradle-api-26.1.2.jar

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s

Finished with error: Please review your Gradle project setup in the android/ folder.

**Comment by maryx** [comment]
Can you try a `flutter clean` and remove your `build/` directory?
You can also try deleting and reinstalling all files related to gradle.
If it's a Gradle issue, sometimes re-installing Android Studio helps, though I know you'd prefer not to do that.

**Comment by alexmeseguer** [comment]
I have tried this, but it always shows the same error. 

I have created a new Flutter Project and the error is the same.

**Comment by alexmeseguer** [comment]
In addition, I've tried create a new app from console but the error is the same:

C:\Users\ameseguer\Desktop>flutter create nueva_flutter_app
Creating project nueva_flutter_app...
  nueva_flutter_app\.gitignore (created)
  nueva_flutter_app\.idea\libraries\Dart_SDK.xml (created)
  nueva_flutter_app\.idea\libraries\Flutter_for_Android.xml (created)
  nueva_flutter_app\.idea\modules.xml (created)
  nueva_flutter_app\.idea\runConfigurations\main_dart.xml (created)
  nueva_flutter_app\.idea\workspace.xml (created)
  nueva_flutter_app\.metadata (created)
  nueva_flutter_app\android\app\build.gradle (created)
  nueva_flutter_app\android\app\src\main\java\com\example\nuevaflutterapp\MainActivity.java (created)
  nueva_flutter_app\android\build.gradle (created)
  nueva_flutter_app\android\app\src\main\AndroidManifest.xml (created)
  nueva_flutter_app\android\app\src\main\res\drawable\launch_background.xml (created)
  nueva_flutter_app\android\app\src\main\res\mipmap-hdpi\ic_launcher.png (created)
  nueva_flutter_app\android\app\src\main\res\mipmap-mdpi\ic_launcher.png (created)
  nueva_flutter_app\android\app\src\main\res\mipmap-xhdpi\ic_launcher.png (created)
  nueva_flutter_app\android\app\src\main\res\mipmap-xxhdpi\ic_launcher.png (created)
  nueva_flutter_app\android\app\src\main\res\mipmap-xxxhdpi\ic_launcher.png (created)
  nueva_flutter_app\android\app\src\main\res\values\styles.xml (created)
  nueva_flutter_app\android\gradle\wrapper\gradle-wrapper.properties (created)
  nueva_flutter_app\android\gradle.properties (created)
  nueva_flutter_app\android\settings.gradle (created)
  nueva_flutter_app\ios\Runner\AppDelegate.h (created)
  nueva_flutter_app\ios\Runner\AppDelegate.m (created)
  nueva_flutter_app\ios\Runner\main.m (created)
  nueva_flutter_app\ios\Runner.xcodeproj\project.pbxproj (created)
  nueva_flutter_app\ios\Flutter\AppFrameworkInfo.plist (created)
  nueva_flutter_app\ios\Flutter\Debug.xcconfig (created)
  nueva_flutter_app\ios\Flutter\Release.xcconfig (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Contents.json (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-1024x1024@1x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-20x20@1x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-20x20@2x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-20x20@3x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-29x29@1x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-29x29@2x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-29x29@3x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-40x40@1x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-40x40@2x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-40x40@3x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-60x60@2x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-60x60@3x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-76x76@1x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-76x76@2x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-83.5x83.5@2x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\Contents.json (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\LaunchImage.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\LaunchImage@2x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\LaunchImage@3x.png (created)
  nueva_flutter_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\README.md (created)
  nueva_flutter_app\ios\Runner\Base.lproj\LaunchScreen.storyboard (created)
  nueva_flutter_app\ios\Runner\Base.lproj\Main.storyboard (created)
  nueva_flutter_app\ios\Runner\Info.plist (created)
  nueva_flutter_app\ios\Runner.xcodeproj\project.xcworkspace\contents.xcworkspacedata (created)
  nueva_flutter_app\ios\Runner.xcodeproj\xcshareddata\xcschemes\Runner.xcscheme (created)
  nueva_flutter_app\ios\Runner.xcworkspace\contents.xcworkspacedata (created)
  nueva_flutter_app\lib\main.dart (created)
  nueva_flutter_app\nueva_flutter_app.iml (created)
  nueva_flutter_app\nueva_flutter_app_android.iml (created)
  nueva_flutter_app\pubspec.yaml (created)
  nueva_flutter_app\README.md (created)
  nueva_flutter_app\test\widget_test.dart (created)
Running "flutter packages get" in nueva_flutter_app...      10,7s
Wrote 63 files.

[√] Flutter is fully installed. (Channel beta, v0.9.4, on Microsoft Windows [VersiÃ³n 10.0.17134.345], locale es-ES)
[!] Android toolchain - develop for Android devices is partially installed; more components are available. (Android SDK 28.0.3)
[√] Android Studio is fully installed. (version 3.2)
[√] VS Code, 64-bit edition is fully installed. (version 1.28.2)
[√] Connected devices is fully installed. (1 available)

Run "flutter doctor" for information about installing additional components.

All done! In order to run your application, type:

  $ cd nueva_flutter_app
  $ flutter run

Your main program file is lib/main.dart in the nueva_flutter_app directory.


C:\Users\ameseguer\Desktop>cd nueva_flutter_app

C:\Users\ameseguer\Desktop\nueva_flutter_app>flutter run
Launching lib/main.dart on Redmi 4 in debug mode...
Initializing gradle...                                       7,8s
Resolving dependencies...
* Error running Gradle:
Exit code 1 from: C:\Users\ameseguer\Desktop\nueva_flutter_app\android\gradlew.bat app:properties:
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details


FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'android'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve com.android.tools.build:gradle:3.1.2.
     Required by:
         project :
      > Could not resolve com.android.tools.build:gradle:3.1.2.
         > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
            > Could not GET 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
               > Connection reset
      > Could not resolve com.android.tools.build:gradle:3.1.2.
         > Could not get resource 'https://jcenter.bintray.com/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
            > Could not GET 'https://jcenter.bintray.com/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
               > Connection reset

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 57s


Please review your Gradle project setup in the android/ folder.


What is the problem?

**Comment by maryx** [comment]
Your `flutter doctor` says:
```
[!] Android toolchain - develop for Android devices is partially installed; more components are available. (Android SDK 28.0.3)
```
You should update that, so that `flutter doctor` outputs check boxes for all of the items. You can search online to see how to do it, and maybe https://github.com/flutter/flutter/issues/16280 can help?

If it still doesn't work after that, you may want to try searching up your Gradle error.

**Comment by alexmeseguer** [comment]
The "Android toolchain" error is already corrected:

`Microsoft Windows [Versión 10.0.17134.345]
(c) 2018 Microsoft Corporation. Todos los derechos reservados.

C:\Users\ameseguer\Desktop\flutter_apm>flutter doctor -v
[√] Flutter (Channel beta, v0.9.4, on Microsoft Windows [VersiÃ³n 10.0.17134.345], locale es-ES)
    • Flutter version 0.9.4 at C:\Apps\Flutter
    • Framework revision f37c235c32 (5 weeks ago), 2018-09-25 17:45:40 -0400
    • Engine revision 74625aed32
    • Dart version 2.1.0-dev.5.0.flutter-a2eb050044

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at C:\Users\ameseguer\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[√] Android Studio (version 3.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 29.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[√] VS Code, 64-bit edition (version 1.28.2)
`

But the dependencies error persists:

`Launching lib\main.dart on Redmi 4 in debug mode...
Initializing gradle...
Resolving dependencies...
timeout waiting for the application to start
* Error running Gradle:
Exit code 1 from: C:\Users\ameseguer\Desktop\flutter_apm\android\gradlew.bat app:properties:
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details


FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'android'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve com.android.tools.build:gradle:3.1.2.
     Required by:
         project :
      > Could not resolve com.android.tools.build:gradle:3.1.2.
         > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
            > Could not GET 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
               > Connect to 30.0.0.142:80 [/30.0.0.142] failed: Connection timed out: connect
      > Could not resolve com.android.tools.build:gradle:3.1.2.
         > Could not get resource 'https://jcenter.bintray.com/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
            > Could not GET 'https://jcenter.bintray.com/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
               > Connect to 30.0.0.142:80 [/30.0.0.142] failed: Connection timed out: connect

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 6s

Finished with error: Please review your Gradle project setup in the android/ folder.
`

**Comment by maryx** [comment]
Is this happening for other Flutter projects? E.g. when you create a new project, or open another project? If so, please file a bug at https://github.com/flutter/flutter/issues. You can link this one there.

If it works for other projects, try re-downloading and re-building a fresh project from scratch.

Also, you could try deleting Android Studio entirely (including all settings files) and downloading it again.

**Comment by htasha** [comment]
same here

**Comment by maryx** [comment]
This might be related to https://github.com/flutter/flutter/issues/23996.

**Comment by iRuth** [comment]
I had the same problem. You need to ensure that your flutter SDK is the latest from https://github.com/flutter/flutter. To fix the issue, from the terminal, go to the folder that has flutter SDK on your computer and run `git pull`.

**Comment by maryx** [comment]
Also see https://github.com/flutter/flutter/issues/23404, it might help.
@alexmeseguer let us know!
If resolved, feel free to close the issue.

**Comment by LouayNasr** [comment]
the SDKversion and build tools version in the build.gradle located inside the **android/app folder**  are 26 and 26.0.3
but you may have another version install in sdkmanager.
so....
you may want to change a few things in the build.gradle file to match your version of android sdk and build tools :
1. compileSdkVersion 26 (change 26 to match your SDKversion)
2. buildToolsVersion '26.0.3'
3. targetSdkVersion 26

and in build.gradle file located in **android folder** _not the same file above_ .. change the gradle version in the dependencies 
(i created new project and and updated the gradle version below to match the one in the new project ) :
`dependencies {
        classpath 'com.android.tools.build:gradle:3.0.1'
    }` 
i hope this will help you.

**Comment by aleksoft1** [comment]
I fix this issue in _task_02_category_widget/android/build.gradle_ update 'com.android.tools.build:gradle:3.0.1' to 'com.android.tools.build:gradle:3.2.1':
`buildscript {
    repositories {
        google()
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.2.1'
    }
}

allprojects {
    repositories {
        google()
        jcenter()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

task clean(type: Delete) {
    delete rootProject.buildDir
}
`

**Comment by maryx** [comment]
This should be resolved with #158. If it is, please feel free to close this issue.
